doc/dev/config.rst

```diff
@@ -57,7 +57,7 @@ Reading configuration values
 There are two ways for Ceph code to get configuration values. One way is to
 read it directly from a variable named "g_conf," or equivalently,
 "g_ceph_ctx->_conf." The other is to register an observer that will be called
-every time the relevant configuration values changes.  This observer will be
+every time the relevant configuration values changes. This observer will be
```

Comment:
@tchaikov I found that these minor changes looks better along with other typos that are fixed by @shun-s in this PR. The mistake I did was: somehow I missed the  `legacy_config_opts.h` revert. Created a new PR for that: https://github.com/ceph/ceph/pull/16721.